include/aos/common/tools/timer.hpp

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_TIMER_HPP_
+#define AOS_TIMER_HPP_
+
+#include <pthread.h>
+#include <signal.h>
+#include <string.h>
+#include <time.h>
+
+#include "aos/common/config.hpp"
+#include "aos/common/tools/function.hpp"
+
+namespace aos {
+/**
+ * Timer instance.
+ * @tparam T timer callback type.
+ */
+class Timer {
+public:
+    /**
+     * Constructs timer instance.
+     */
+    Timer() = default;
+
+    /**
+     * Destructs timer instance.
+     */
+    ~Timer() { Stop(); }
+
+    /**
+     * Creates timer.
+     *
+     * @param intervalMs Timer interval in milliseconds.
+     *
+     * @return Error code.
+     */
+    template <typename F>
+    Error Create(unsigned int intervalMs, F functor, void* arg = nullptr)
+    {
+        if (mTimerID != 0) {
+            auto err = Stop();
+            if (!err.IsNone()) {
+                return err;
+            }
+        }
+
+        auto err = mFunction.Capture(functor, arg);
+        if (!err.IsNone()) {
+            return err;
+        }
+
+        mIntervalMs = intervalMs;
+
+        struct sigevent   sev { };
+        struct itimerspec its { };
+
+        sev.sigev_notify = cTimerSigevNotify;
+        sev.sigev_value.sival_ptr = this;
+        sev.sigev_notify_function = TimerFunction;
+
+        its.it_value.tv_sec = intervalMs / 1000;
+        its.it_value.tv_nsec = (intervalMs % 1000) * 1000000;
+
+        auto ret = timer_create(CLOCK_MONOTONIC, &sev, &mTimerID);
+        if (ret != 0) {
+            return ret;
```

Comment:
According to documentaion `timer_create` returns -1 in error case. I guess, errno should be returned.

Reply:
Done